src/scripts/main.js

```diff
@@ -0,0 +1,120 @@
+const formContainer = document.querySelector('.form__container');
+const addButton = document.querySelector('.form__button--add');
+const submitButton = document.querySelector('.form__button--submit');
+const MAX_INPUT_COUNT = 6;
+
+const regEx = /.+?(?=[?]passwords=)/;
+
+let activeInputs = 0;
+
+const checkInputs = inputs => {
+    const emptyInputs = Array.from(inputs).filter(input => input.value === '');
+    if (emptyInputs.length > 0) {
+        const errorParagraph = /* HTML */ `
+            <p class="form__error">The field cannot be empty</p>
+        `;
+        emptyInputs.forEach(emptyInput =>
+            emptyInput.parentNode.insertAdjacentHTML('afterend', errorParagraph)
+        );
+        return false;
+    }
+    return true;
+};
+
+const deleteErrorParagraphs = () => {
+    const formSections = document.querySelectorAll('.form__section');
+    formSections.forEach(formSection => {
+        if (formSection.querySelector('.form__error')) {
+            const formError = formSection.querySelector('.form__error');
+            formSection.removeChild(formError);
+        }
+    });
+};
+
+const updateCoutner = result => {
+    if (result) {
+        activeInputs += 1;
+    } else {
+        activeInputs -= 1;
+    }
+};
+
+const updateNumbers = () => {
+    const spans = document.querySelectorAll('.form__number');
+    spans.forEach((span, index) => (span.textContent = index + 1));
+};
+
+const generateInput = activeInputs => {
+    let formDeleteButton = '';
+    let defaultInputText = '';
+    if (activeInputs > 0) {
+        formDeleteButton = /* HTML */ `
+            <button class="form__delete" type="button"></button>
+        `;
+    } else {
+        defaultInputText = 'It is default input text.';
+    }
+    console.log(defaultInputText);
+    const formInput = /* HTML */ `
```

Comment:
do osobnej funkcji najlepiej to przenieś z argumentami: `renderInput(activeInputs, defaultInputText, formDeleteButton)`